### PR TITLE
Log as cmo service account in make run-local script

### PR DESCRIPTION
**Pull request by marioferh**
After some issues with cert creation:
 {[{Denied True CSRDenied CSR "system:openshift:openshift-monitoring-dvpqj" was created by an unexpected user: "system:admin"

This change allows correct cmo sa to test changes in local: make run-local

Add: 
make run-local-as-cmo

* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: hack/local-cmo.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 set -eu -o pipefail
+set -x
+set -v
```
Comment (sthaha) [inline hack/local-cmo.sh:4]: -x is too verbose IMHO and hinders readability.

File: assets/prometheus-k8s/prometheus.yaml [hunk 2]
```diff
@@ -82,7 +82,7 @@ spec:
   - args:
     - --secure-listen-address=0.0.0.0:9092
     - --upstream=http://127.0.0.1:9090
-    - --allow-paths=/metrics
+    - --allow-paths=/metrics,/federate
```
Comment (sthaha) [inline assets/prometheus-k8s/prometheus.yaml:85]: Unrelated change?

File: hack/local-cmo.sh [hunk 3]
```diff
@@ -78,12 +80,40 @@ images_from_deployment() {
   kc get deployment cluster-monitoring-operator -o json | jq -r '.spec.template.spec.containers[] | select(.name=="cluster-monitoring-operator") | .args[] | select(.|test("\\-images.*"))'
 }
 
+#log as cmo sa in order to be the correct user
+kc_cmo_login() {
+	local cmo_sa="system:serviceaccount:openshift-monitoring:cluster-monitoring-operator"
+	local tmp_dir
+	tmp_dir="$(git rev-parse --show-toplevel)/tmp/$(git branch --show-current)"
```
Comment (sthaha) [inline hack/local-cmo.sh:87]: I think we can use "PROJECT_DIR/tmp" itself for this.

File: assets/prometheus-k8s/prometheus.yaml (hunk 2 above)
Comment (marioferh) [reply assets/prometheus-k8s/prometheus.yaml:85]: yep sorry I was working above another change :S

File: hack/local-cmo.sh (hunk 1 above)
Comment (marioferh) [reply hack/local-cmo.sh:4]: Yep, it was for test purposes only

File: jsonnet/components/prometheus.libsonnet [hunk 4]
```diff
@@ -203,7 +203,25 @@ function(params)
       data: {},
     },
 
-    kubeRbacProxySecret: generateSecret.staticAuthSecret(cfg.namespace, cfg.commonLabels, 'kube-rbac-proxy'),
+    kubeRbacProxySecret: generateSecret.staticAuthSecret(
```
Comment (sthaha) [inline jsonnet/components/prometheus.libsonnet:206]: Unrelated patch?

Comment (marioferh) [reply jsonnet/components/prometheus.libsonnet:206]: Yep

File: hack/local-cmo.sh [hunk 5]
```diff
@@ -78,12 +80,40 @@ images_from_deployment() {
...
 run() {
   echo "Running: $*"
   "$@"
 }
 
 main(){
+  echo "log as cmo sa?"
+  echo "1.- Yes, remember: oc login -u system:admin when finished"
+  echo "2.- continue as admin"
+  read select
```
[the remark this answers is not in the document]
Comment (marioferh) [reply hack/local-cmo.sh:115]: You are right, not so clear but is here: https://github.com/openshift/cluster-monitoring-operator/blob/43ffa9e515659e02a313ff504e308fb5ac0c17a5/hack/local-cmo.sh#L133
I will improve it

File: hack/local-cmo.sh [hunk 6]
```diff
@@ -1,12 +1,36 @@
...
+    -n|--no-cmo-login)
+      cmo_login='false'
+      shift
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
```
Comment (sthaha) [inline hack/local-cmo.sh:33]: Could we please move this to a parse_args function? The rule of thumb for maintainable scripts is to keep as much as possible in a function so that there aren't any adhoc execution and the flow is predictable.

File: hack/local-cmo.sh [hunk 7]
```diff
@@ -100,6 +148,12 @@ main(){
   local -a images
   while read -r img; do images+=( "$img" ); done < <(images_from_deployment)
 
+  if [ $cmo_login == 'true' ]; then
+    echo "loggin as cmo sa"
+    echo "run: oc login -u system:admin when finished in order to continue as admin"
+    login_as_cmo
+  fi
```
Comment (sthaha) [inline hack/local-cmo.sh:155]:
```suggestion
  if  $cmo_login; then
    echo "logging as cmo sa"
    login_as_cmo
    echo "run: oc login -u system:admin when finished in order to continue as admin"
  fi
```

File: hack/local-cmo.sh [hunk 8]
```diff
@@ -1,12 +1,36 @@
...
+    -h|--help)
+      echo "local_cmo [options]"
+      echo " "
+      echo "options:"
+      echo "-h, --help                show brief help"
+      echo "-n, --no-cmo-login        run script as kube:admin"
+      exit 0
+      ;;
+    -n|--no-cmo-login)
+      cmo_login='false'
```
Comment (sthaha) [inline hack/local-cmo.sh:25]:
```suggestion
      CMO_LOGIN=false
```

File: hack/local-cmo.sh [hunk 9]
```diff
@@ -1,12 +1,36 @@
 #!/usr/bin/env bash
 set -eu -o pipefail
 
+cmo_login='true'
```
Comment (sthaha) [inline hack/local-cmo.sh:4]:
```suggestion
declare CMO_LOGIN=true
```

File: hack/local-cmo.sh (hunk 7 above)
Comment (sthaha) [reply hack/local-cmo.sh:155]: I think we can improve the script by switching back to the old use when the script exits `trap restore_login EXIT`

**Comment by marioferh** [comment]
closing this to continue with: https://github.com/openshift/cluster-monitoring-operator/pull/2010
